Main.py

```diff
@@ -306,13 +306,27 @@ def find_common_pool(players: Set[int], shared_pool: Set[str]) -> Tuple[
     else:
         logger.info("Progression balancing skipped.")
 
-    # we're about to output using multithreading, so we're removing the global random state to prevent accidental use
-    multiworld.random.passthrough = False
-
     if args.skip_output:
         logger.info('Done. Skipped output/spoiler generation. Total Time: %s', time.perf_counter() - start)
         return multiworld
 
+    # determine which items should get starting hints and add those location to start_location_hints to be collected
+    # into multidata later
+    filled_locations = multiworld.get_filled_locations()
+    multiworld.random.shuffle(filled_locations)
+    collected_hints = {player: collections.Counter() for player in multiworld.get_all_ids()}
+    for location in filled_locations:
+        if type(location.address) == int:
+            if location.item.name in multiworld.worlds[location.item.player].options.start_hints:
+                item = location.item
+                if (collected_hints[item.player].setdefault(item.name, 0)
```

Comment:
The setdefault isn't necessary here right? A Counter I believe already acts like a defaultdict with lambda: 0. I tested it and it didn't break at the very least